filters/openpolicyagent/opaauthorizerequest/benchmark_test.go

```diff
@@ -0,0 +1,421 @@
+package opaauthorizerequest
+
+import (
+	"fmt"
+	"github.com/golang-jwt/jwt/v4"
+	opasdktest "github.com/open-policy-agent/opa/sdk/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/filters/filtertest"
+	"github.com/zalando/skipper/filters/openpolicyagent"
+	"github.com/zalando/skipper/metrics/metricstest"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"strings"
+	"testing"
+	"time"
+)
+
+const (
+	tokenExp = 2 * time.Hour
+
+	certPath = "../../../skptesting/cert.pem"
+	keyPath  = "../../../skptesting/key.pem"
+
+	testDecisionPath = "envoy/authz/allow"
+	testBundleName   = "somebundle.tar.gz"
+)
+
+var testBundleEndpoint = fmt.Sprintf("/bundles/%s", testBundleName)
+
+func BenchmarkMinimalPolicy(b *testing.B) {
+	opaControlPlane := opasdktest.MustNewServer(
+		opasdktest.MockBundle(testBundleEndpoint, map[string]string{
+			"main.rego": `
+					package envoy.authz
+
+					default allow = false
+
+					allow {
+						input.parsed_path = [ "allow" ]
+					}
+				`,
+		}),
+	)
+
+	filterOpts := NewFilterOptionsWithDefaults(opaControlPlane.URL())
+	f, err := createOpaFilter(filterOpts)
+	require.NoError(b, err)
+
+	reqUrl, err := url.Parse("http://opa-authorized.test/allow")
+	require.NoError(b, err)
+
+	ctx := &filtertest.Context{
+		FStateBag: map[string]interface{}{},
+		FResponse: &http.Response{},
+		FRequest: &http.Request{
+			URL: reqUrl,
+		},
+		FMetrics: &metricstest.MockMetrics{},
+	}
+
+	b.ResetTimer()
+	b.RunParallel(func(pb *testing.PB) {
```

Comment:
Might be redundant to general docs for benchmarks but I'd recommend to add the command line with which you run this benchmark and how you control the parallelism. This would help anybody to create reproducible results